asv_bench/benchmarks/pandas/join_merge.py

```diff
@@ -0,0 +1,261 @@
+import string
```

Comment:
We already have join, merge and concat benchmarks in place. let's try to add maximum 1-2 runs in each case. @YarShev 
 @anmyachev I think it's your call :)

Reply:
I am in favor of leaving as is, those functions that you have named, as they are one of the most important. A little redundancy can be dealt with later.

Now the main thing is to enter these benchmarks into the Modin system. We need to do a couple of things for this:
* add test runs of these benchmarks to CI
* add runs of these benchmarks to our main benchmarking configuration
* use the functionality from this pull request so that the code can work in two modes. https://github.com/modin-project/modin/pull/5015
* so far, there is no automatic mechanism implemented that could automatically trigger all calculations so that we get the correct times, which means it will need to be done manually (`BenchmarkMode` does not work for OmniSci and besides, it can slow down the `setup` function)
* in addition, we need to solve the issue with the dimensions of the datasets. At too small dimensions, where pandas can execute a function in a few tens of milliseconds, modin is unlikely to be able to overtake pandas. However, if you increase the dataset, you may encounter a lack of memory on the machine that runs these validation tests during the CI process. Because of this, the already added benchmarks have a built-in functionality that allows you to change the data size for benchmarks depending on these situations, but it is still manual, we need to manually select the size for these two situations.

Reply:
> I am in favor of leaving as is, those functions that you have named, as they are one of the most important. A little redundancy can be dealt with later.


By "leaving as is" you mean current asv testing or contents of this PR? :) I agree that functions are very important but it could be too heavy for our CI to run all cases from PR

Reply:
I mean content of this PR.

Reply:
Okay, let's try and see it goes but still not sure we want (at least now) `# outer join of non-unique` case and all cases from Merge (eg empty and cross)

Reply:
> Now the main thing is to enter these benchmarks into the Modin system

@anmyachev, could you help @jbrockmendel to start doing this?